daml-script/dump/src/test/scala/com/daml/script/dump/EncodeCreatedSpec.scala

```diff
@@ -11,51 +11,77 @@ import org.scalatest.matchers.should.Matchers
+          fields = Seq.empty,
+        )
+      ),
+    )
   "encodeCreatedEvent" - {
     "contract id bindings" - {
       "unreferenced" in {
         val parties = Map(Party("Alice") -> "alice_0")
-        val cidMap = Map(ContractId("cid") -> "contract_0_0")
+        val cidMap = Map(ContractId("cid1") -> "contract_0_0")
         val cidRefs = Set.empty[ContractId]
-        val created = CreatedEvent(
-          eventId = "create",
-          templateId = Some(Identifier("package", "Module", "Template")),
-          contractId = "cid",
-          signatories = Seq("Alice"),
-          createArguments = Some(
-            Record(
-              recordId = Some(Identifier("package", "Module", "Template")),
-              fields = Seq.empty,
-            )
-          ),
-        )
-        encodeCreatedEvent(parties, cidMap, cidRefs, created).render(80) shouldBe
+        val created = mkCreated(1)
+        encodeSubmitCreatedEvents(parties, cidMap, cidRefs, Seq(created)).render(80) shouldBe
           """submitMulti [alice_0] [] do
             |  createCmd Module.Template""".stripMargin.replace("\r\n", "\n")
       }
       "referenced" in {
         val parties = Map(Party("Alice") -> "alice_0")
-        val cidMap = Map(ContractId("cid") -> "contract_0_0")
-        val cidRefs = Set(ContractId("cid"))
-        val created = CreatedEvent(
-          eventId = "create",
-          templateId = Some(Identifier("package", "Module", "Template")),
-          contractId = "cid",
-          signatories = Seq("Alice"),
-          createArguments = Some(
-            Record(
-              recordId = Some(Identifier("package", "Module", "Template")),
-              fields = Seq.empty,
-            )
-          ),
-        )
-        encodeCreatedEvent(parties, cidMap, cidRefs, created).render(80) shouldBe
+        val cidMap = Map(ContractId("cid1") -> "contract_0_0")
+        val cidRefs = Set(ContractId("cid1"))
+        val created = mkCreated(1)
+        encodeSubmitCreatedEvents(parties, cidMap, cidRefs, Seq(created)).render(80) shouldBe
           """contract_0_0 <- submitMulti [alice_0] [] do
             |  createCmd Module.Template""".stripMargin.replace(
             "\r\n",
             "\n",
           )
       }
+      "multiple referenced" in {
+        val parties = Map(Party("Alice") -> "alice_0")
+        val cidMap = Map(
+          ContractId("cid1") -> "contract_0_0",
+          ContractId("cid2") -> "contract_0_1",
+          ContractId("cid3") -> "contract_0_2",
+        )
+        val cidRefs = Set(ContractId("cid1"), ContractId("cid3"))
+        val events = Seq(mkCreated(1), mkCreated(2), mkCreated(3))
+        encodeSubmitCreatedEvents(parties, cidMap, cidRefs, events).render(80) shouldBe
+          """(contract_0_0, contract_0_2) <- submitMulti [alice_0] [] do
+            |  contract_0_0 <- createCmd Module.Template
+            |  createCmd Module.Template
```

Comment:
should we encode this as `_ <- createCmd …`? I don’t think we turn this warning on by default but it’s generated code so we might as well generate some that doesn’t trigger it.